txmgr: Allow CLI Default values to be configured

trianglesphere opened the pull request:
**Description**

This lets the challenger change it's default CLI configs. The challenger has different needs from the batcher. Specifically, retries in transactions from the challenger are not as consequential or expensive as duplicating txns in the batcher. The challenger also has very good internal retries.

**Metadata**

- Fixes https://github.com/ethereum-optimism/client-pod/issues/61

trianglesphere commented on op-service/txmgr/cli.go:
```diff
@@ -48,17 +48,42 @@ var (
+		TxNotInMempoolTimeout:     2 * time.Minute,
+		ReceiptQueryInterval:      12 * time.Second,
+	}
+	DefaultChallengerFlagValues = DefaultFlagValues{
```
cc @ajsutton for the new values

ajsutton reviewed (approved):
LGTM but you'll need to fix the tests. I suspect the new defaults just need to be used in challengers config.NewConfig

trianglesphere committed:
txmgr: Allow CLI Default values to be configured

This lets the challenger change it's default CLI configs. The challenger has
different needs from the batcher. Specifically, retries in transactions from
the challenger are not as consequential or expensive as duplicating txns in
the batcher. The challenger also has very good internal retries.

trianglesphere commented:
Tests should be fixed now

mergify[bot] committed:
Merge branch 'develop' into jg/configurable_txmgr_cli_default_values

ajsutton commented:
Just pulling this out of the merge queue briefly to get the flaky test fix in.  Then will pull that in here and the tests should pass...

ajsutton committed:
Merge remote-tracking branch 'origin/develop' into jg/configurable_txmgr_cli_default_values

mergify[bot] committed:
Merge branch 'develop' into jg/configurable_txmgr_cli_default_values
